modules/javafx.swing/src/main/java/javafx/embed/swing/InputMethodSupport.java

```diff
@@ -56,42 +58,64 @@ public InputMethodRequestsAdapter(javafx.scene.input.InputMethodRequests fxReque
             this.fxRequests = fxRequests;
         }
 
+        private Point2D pointValue;
+        private int     intValue;
+        private String  stringValue;
+
         @Override
         public Rectangle getTextLocation(TextHitInfo offset) {
-            Point2D result = fxRequests.getTextLocation(offset.getInsertionIndex());
-            return new Rectangle((int)result.getX(), (int)result.getY(), 0, 0);
+            pointValue = new Point2D(0.0, 0.0);
+            if (fxRequests != null) {
+                PlatformImpl.runAndWait(() -> {
+                    pointValue = fxRequests.getTextLocation(offset.getInsertionIndex());
+                });
+            }
+            return new Rectangle((int)pointValue.getX(), (int)pointValue.getY(), 0, 0);
         }
 
         @Override
         public TextHitInfo getLocationOffset(int x, int y) {
-            int result = fxRequests.getLocationOffset(x, y);
-            return TextHitInfo.afterOffset(result);
+            intValue = 0;
+            if (fxRequests != null) {
+                PlatformImpl.runAndWait(() -> {
+                    intValue = fxRequests.getLocationOffset(x, y);
+                });
+            }
+            return TextHitInfo.afterOffset(intValue);
         }
 
         @Override
         public int getInsertPositionOffset() {
+            intValue = 0;
             if (fxRequests instanceof ExtendedInputMethodRequests) {
-                return ((ExtendedInputMethodRequests)fxRequests).getInsertPositionOffset();
+                PlatformImpl.runAndWait(() -> {
+                    intValue = ((ExtendedInputMethodRequests)fxRequests).getInsertPositionOffset();
+                });
             }
-            return 0;
+            return intValue;
         }
 
         @Override
         public AttributedCharacterIterator getCommittedText(int beginIndex, int endIndex, AttributedCharacterIterator.Attribute[] attributes) {
-            String result = null;
+            stringValue = null;
             if (fxRequests instanceof ExtendedInputMethodRequests) {
-                result = ((ExtendedInputMethodRequests)fxRequests).getCommittedText(beginIndex, endIndex);
+                PlatformImpl.runAndWait(() -> {
+                    stringValue = ((ExtendedInputMethodRequests)fxRequests).getCommittedText(beginIndex, endIndex);
+                });
             }
-            if (result == null) result = "";
-            return new AttributedString(result).getIterator();
+            if (stringValue == null) stringValue = "";
```

Comment:
Not sure of FX coding guidelines but in JDK, even for a single statement, we need to put in braces

Reply:
I guess this should add [JDK-8090267](https://bugs.openjdk.org/browse/JDK-8090267) through "/issue add" command

Reply:
Regarding FX coding style, the one place where braces are not needed is when everything is all on one line.

Regarding [JDK-8090267](https://bugs.openjdk.org/browse/JDK-8090267), should the call to  `getInputMethodRequests` added to [JFXPanel line 1028](https://github.com/openjdk/jfx/pull/1169/files#diff-e6f5782dba36f256cceb649ce526ed1f7a8f6704d3a32bd8bd17cef42621d99cR1028) by PR #1169 be added to this PR as well?

Reply:
/issue add JDK-8090267